fix: LEAP-921: Fix the way of updating users in store

Gondragos opened the pull request:
This PR deprecates using of `mergeUsers` and adding `enrichUsers` instead. 

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)


#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
This was done, 'cause `mergeUsers` didn't update existing users and could lose some changes that should be applied. This is actual when we already have list of users that includes email and names and then we try to add more information or some new version of it, like an added `avatar` field.

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
`Comments`, `Users`

Gondragos committed:
fix: LEAP-921: Fix the way of updating users in store

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8817147602

hlomzik commented on web/libs/editor/src/stores/AppStore.js:
```diff
@@ -859,10 +859,25 @@ export default types
+        oldUsersMap[user.id] = user;
+      });
+      const newUsers = users.map(user => {
+        const oldUser = oldUsersMap[user.id];
+
+        return oldUser ? { ...oldUser, ...user } : user;
```
This can be simplified to not create too much extra checks and vars
```suggestion
        return { ...oldUsersMap[user.id], ...user };
```

Gondragos committed:
Refactoring

Co-authored-by: hlomzik <hlomzik@gmail.com>

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8818465294

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8818845098)
> Successfully merged:  1 file changed, 1 insertion(+), 1 deletion(-)

Gondragos committed:
Merge branch 'develop' into 'fb-leap-921/annotation-history'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8818845098

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8818881385

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8880854098)
> Successfully merged:  1132 files changed, 60025 insertions(+), 61739 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-leap-921/annotation-history'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8880854098

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8880889266

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8931406199)
> Successfully merged:  31 files changed, 2806 insertions(+), 266 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-leap-921/annotation-history'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8931406199

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8931422362

Gondragos committed:
Apply biome check corrections

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8941199083
